doc/source/whatsnew/v2.2.0.rst

```diff
@@ -235,6 +235,7 @@ Other Deprecations
 ^^^^^^^^^^^^^^^^^^
 - Changed :meth:`Timedelta.resolution_string` to return ``h``, ``min``, ``s``, ``ms``, ``us``, and ``ns`` instead of ``H``, ``T``, ``S``, ``L``, ``U``, and ``N``, for compatibility with respective deprecations in frequency aliases (:issue:`52536`)
 - Deprecated :meth:`Index.format`, use ``index.astype(str)`` or ``index.map(formatter)`` instead (:issue:`55413`)
+- Deprecated :meth:`Series.to_period`, :meth:`DataFrame.to_period`, :meth:`Series.to_timestamp`, :meth:`DataFrame.to_timestamp`, :meth:`Series.tz_localize`, :meth:`DataFrame.tz_localize`, :meth:`Series.tz_convert`, :meth:`DataFrame.tz_convert`, use the relevant methods on ``obj.set_axis(obj.index.relevant_method(...))`` instead (:issue:`52110`)
```

Comment:
> obj.set_axis(obj.index.relevant_method(...))

I realized this might be inconvenient for users calling these methods in a chain. Maybe we need that `pd.col` or equivalent to make this deprecation more migrate-able?

Reply:
that was part of the appeal of the `obj.axis_ops.tz_convert` idea

Reply:
I wasn't opposed to the `axis_op` idea (I didn't have much time to review the PR sadly)

Reply:
ill revive that PR as an alternative